src/cairo_interaction.rs

```diff
@@ -25,16 +25,16 @@ impl GdkSurfaceExt for Surface {
 }
 
 pub trait GdkPixbufExt {
-    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: &W) -> Option<Surface>;
+    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: Option<&W>) -> Option<Surface>;
 }
 
 impl GdkPixbufExt for Pixbuf {
-    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: &W) -> Option<Surface> {
+    fn create_surface<W: IsA<Window>>(&self, scale: i32, for_window: Option<&W>) -> Option<Surface> {
         unsafe {
             from_glib_full(gdk_sys::gdk_cairo_surface_create_from_pixbuf(
                 self.to_glib_none().0,
                 scale,
-                for_window.as_ref().to_glib_none().0,
+                for_window.to_glib_none().0,
```

Comment:
Seems is neededed something like `.map(std::convert::AsRef::as_ref)` co convert type
```
error[E0308]: mismatched types
  --> src\cairo_interaction.rs:37:17
   |
37 |                 for_window.to_glib_none().0,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `gdk_sys::GdkWindow`, found associated type
   |
   = note: expected type `*mut gdk_sys::GdkWindow`
              found type `*mut <W as glib::ObjectType>::GlibType`
   = note: consider constraining the associated type `<W as glib::ObjectType>::GlibType` to `gdk_sys::GdkWindow`
   = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
error: aborting due to previous error
```

And some `cargo fmt` too

Reply:
Thanks, should be fixed now